### DISCUSSION TITLE
How much time does the cache need to reset

### DISCUSSION THREAD

**Discussion by PhilipAbed** [sidorares/node-mysql2#1605] [Q&A] [answered]
in docs it says that if you run same query twice, u will get the results from LRU cache, 
1) is there a way to cancel caching?
2) if no, how much time is it cached for?

**Comment by sidorares** [comment, marked as answer]
1) currently there is no way to opt out. There is `maxPreparedStatements` connection parameter, you can set it to a low number to control how many statements single connection is used. Note that it does not give a 100% guarantee - in mysql protocol statement_close is a one way message and there is no way to know when statement is actually closed. Not sure what is LRU behaviour when you set maxPreparedStatements to 0 - this might be a way for you to "cancel caching" but needs checking. Another option would be using manual `cont stmt = connection.prepare(sql); ... statement.execute(parameters); ...statement.close()`. This should be not cached ( need to confirm that as well - see discussion in https://github.com/sidorares/node-mysql2/issues/805#issuecomment-447871245 )
2) it's cached until the connection is closed, or reset ( this is future functionality not there yet - #1437 ) or [evicted](https://github.com/sidorares/node-mysql2/blob/11879df6e387f0d7bacd6d0a6898d10d7ac2274b/lib/connection.js#L74) because number of statements in a given connection is more than `maxPreparedStatements`. Currently there is no time based eviction, and the cache we use does not support accurate time based eviction (and there is a performance hit if we enable `ttlAutopurge` )  :

> https://www.npmjs.com/package/lru-cache
This is not primarily a TTL cache, and does not make strong TTL guarantees. There is no preemptive pruning of expired items by default, but you may set a TTL on the cache or on a single set. If you do so, it will treat expired items as missing, and delete them when fetched. If you are more interested in TTL caching than LRU caching, check out [@isaacs/ttlcache](http://npm.im/@isaacs/ttlcache).

**Reply by PhilipAbed** [reply]
Thanks for the fast reply!
That doesn't help me much though, ill just work around it somehow.

**Reply by sidorares** [reply]
What's the issue you are fixing?

**Reply by PhilipAbed** [reply]
im using github graphql to get info from github into my database,
 so i can calculate statistics and stuff, 
but i also need to update the database with those infos on daily/weekly bases

i was testing the code, suddenly the Insert is throwing an error, and apparently the Select didnt return the rows i just inserted, so i end up inserting duplicate keys.
i think it's fine though since we will run this daily it shouldn't be a problem 

p.s. 
i noticed u have some outdated dependencies, if you wonder what i do, 
you could use https://github.com/renovatebot/renovate
free automatic dependency update tool, enjoy
